src/formio/components/SoftRequiredErrors.js

```diff
@@ -0,0 +1,59 @@
+import FormioUtils from 'formiojs/utils';
+import {Formio} from 'react-formio';
+
+const FormioContentField = Formio.Components.components.content;
+
+class SoftRequiredErrors extends FormioContentField {
+  static schema(...extend) {
+    return FormioContentField.schema(
+      {
+        type: 'softRequiredErrors',
+        key: 'softRequiredErrors',
+        label: '', // not displayed anyway
+      },
+      ...extend
+    );
+  }
+
+  constructor(component, options, data) {
+    component.refreshOnChange = true;
+    super(component, options, data);
+  }
+
+  get content() {
+    // figure out which components use soft required validation
+    const softRequiredComponents = [];
+    FormioUtils.eachComponent(this.root.components, component => {
+      if (component.component.openForms?.softRequired) {
+        softRequiredComponents.push(component);
+      }
+    });
+
+    const missingFieldLabels = [];
+    // check which components have an empty value
+    for (const component of softRequiredComponents) {
+      const isEmpty = component.isEmpty();
+      if (isEmpty) missingFieldLabels.push(component.label);
+    }
+
+    if (!missingFieldLabels.length) return '';
+
+    const missingFieldsMarkup = this.renderTemplate('missingFields', {labels: missingFieldLabels});
+    const content = this.interpolate(this.component.html, {
+      missingFields: missingFieldsMarkup,
+    });
+
+    return `
+      <div id="${this.id}-content" class="utrecht-alert utrecht-alert--warning openforms-soft-required-errors">
```

Comment:
To add some context for users with a visual disability:

```suggestion
      <div id="${this.id}-content" class="utrecht-alert utrecht-alert--warning openforms-soft-required-errors" role="alert">
```

Reply:
From https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role:

> The alert role is for important, and usually time-sensitive, information.

I don't find this time sensitive, especially because it's also displayed at the beginning of a form step that is yet to be filled out.

> The alert role should only be used for information that requires the user's immediate attention, for example:

This is not the case - it becomes important once you are about to submit the form.

I think the `role="status"` is more appropriate: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/status_role